File: intToRoman.py
class Solution: 
    def intToRoman(self, num: int) -> str:
        digits = [
            (1000, "M"),
            (900, "CM"),
            (500, "D"),
            (400, "CD"),
            (100, "C"),
            (90, "XC"),
            (50, "L"),
            (40, "XL"),
            (10, "X"),
            (9, "IX"),
            (5, "V"),
            (4, "IV"),
            (1, "I"),
        ]
        ans = ""
        for curr, symbol in digits:
            if num == 0:
                break
            if num >= curr:
                count, num = divmod(num, curr)
                ans += (symbol * count)
        return ans

File: intToRoman.js
var intToRoman = function(num) {
    const values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
    const symbols = [
        "M",
        "CM",
        "D",
        "CD",
        "C",
        "XC",
        "L",
        "XL",
        "X",
        "IX",
        "V",
        "IV",
        "I",
    ];
    ans = "";
    for (let i = 0; i < values.length && num > 0; i++) {
        if (num >= values[i]) {
            let repeat = Math.floor(num/values[i]);
            for (let j = 0; j < repeat; j++) {
                ans += symbols[i];
            }
            num %= values[i];
        }
    }
    return ans;

}

console.log(intToRoman(3749));